migrations/helpers/dart_samples_update_helper.py

```diff
@@ -0,0 +1,387 @@
+
+            print(f"Selecting positive samples between {s_start_datetime} and {s_end_datetime}")
+
+            # 1. get samples from mongo between these time ranges that are positive
+            first_match = {
+                "$match": {
+                    # 1. First filter by the start and end dates
+                    FIELD_CREATED_AT: {"$gte": start_datetime, "$lte": end_datetime},
+                    FIELD_RESULT: {"$regex": "^positive", "$options": "i"},
+                    FIELD_ROOT_SAMPLE_ID: {"$not": {"$regex": "^CBIQA_"}},
+                }
+            }
+
+            pipeline = [first_match]
+
+            results = samples_collection.aggregate(pipeline)
+
+            num_docs = 0
+
+            root_sample_ids = []
+            plate_barcodes = set()
+
+            for doc in results:
+                root_sample_ids.append(doc[FIELD_ROOT_SAMPLE_ID])
+                plate_barcodes.add(doc[FIELD_PLATE_BARCODE])
+
+                num_docs += 1
+
+            print(f"{num_docs} to process")
+            print(f"{len(plate_barcodes)} unique plate barcodes")
+
+            # 2. of these, find which have been cherry-picked and remove them from the list
+            cp_samples_df = get_cherrypicked_samples(config, root_sample_ids, plate_barcodes)
+
+            # create a second match for the pipeline
+            second_match = {
+                "$match": {
+                    FIELD_ROOT_SAMPLE_ID: {"$nin": cp_samples_df[FIELD_ROOT_SAMPLE_ID].to_list()},  # type: ignore
+                }
+            }
+
+            pipeline.append(second_match)  # type: ignore
+
+            # get the samples between those dates minus the cherry-picked ones
+            results = samples_collection.aggregate(pipeline)
+            samples = list(results)
+
+            print(
+                f"{len(samples)} documents found in the mongo database between these timestamps and not cherry-picked"
+            )
+
+            # 3. add the relevant filtered positive fields if not present
+            filtered_positive_identifier = FilteredPositiveIdentifier()
+            version = filtered_positive_identifier.current_version()
+            update_timestamp = datetime.now()
+
+            print("Updating filtered positives...")
+
+            update_filtered_positive_fields(
+                filtered_positive_identifier,
+                samples,
+                version,
+                update_timestamp,
+            )
+
+            print("Updated filtered positives")
+
+            # 4. add the UUID fields if not present
+            add_sample_uuid_field(samples)
+
+            samples_updated_with_source_plate_uuids(mongo_db, samples)
+
+            # 5. update samples in mongo updated in either of the above two steps (would expect the same set of samples
+            #       from both steps)
+            print("Updating Mongo...")
+            _ = update_mongo_fields(mongo_db, samples, version, update_timestamp)
```

Comment:
Would it not be best to do a "if mongo fails, don't update MLWH or DART; if MLWH fails, don't update DART" workflow? Or am I missing that this is already in place (with exceptions?)?

Reply:
Yes, I need to implement that check, thanks for the reminder.

Reply:
Actually, any failure should throw an exception and stop but will add a few raises to double check my understanding.

Reply:
Let's update the migration to use the enhancement required in #162 .